pkg/clamav/scanner.go

```diff
@@ -0,0 +1,60 @@
+package clamav
+
+import (
+	"bytes"
+	"fmt"
+
+	clamd "github.com/dutchcoders/go-clamd"
+	"github.com/fsouza/go-dockerclient"
+
+	"github.com/openshift/image-inspector/pkg/api"
+)
+
+type ClamScanner struct {
+	// Socket is the location of the clamav socket.
+	Socket string
+}
+
+var _ api.Scanner = &ClamScanner{}
+
+func NewScanner(socket string) api.Scanner {
+	return &ClamScanner{
+		Socket: socket,
+	}
+}
+
+// Scan will scan the image
+func (s *ClamScanner) Scan(path string, image *docker.Image) error {
+	scanner := clamd.NewClamd(s.Socket)
```

Comment:
probably the scanner interface have Validate() method to check if you have all requirements needed for running scan?